.gitignore

```diff
@@ -19,3 +22,184 @@ doc/api/
 *.js_
 *.js.deps
 *.js.map
+
+.flutter-plugins
+.flutter-plugins-dependencies
+
+# General
+.DS_Store
+.AppleDouble
+.LSOverride
+
+# Icon must end with two \r
+Icon
+
+# Thumbnails
+._*
+
+# Files that might appear in the root of a volume
+.DocumentRevisions-V100
+.fseventsd
+.Spotlight-V100
+.TemporaryItems
+.Trashes
+.VolumeIcon.icns
+.com.apple.timemachine.donotpresent
+
+# Directories potentially created on remote AFP share
+.AppleDB
+.AppleDesktop
+Network Trash Folder
+Temporary Items
+.apdisk
+
+# Windows thumbnail cache files
+Thumbs.db
+Thumbs.db:encryptable
+ehthumbs.db
+ehthumbs_vista.db
+
+# Dump file
+*.stackdump
```

Comment:
why would we need all of this?

Reply:
This is based on templates from https://github.com/github/gitignore. The flutter one is oversimplified - missing a lot of OS and editor rules to common not to commit files. For example, nothing about Windows and network file system files.

Reply:
Yeah but I mean, we are on a simple Flutter project here, not a huge repo like linux or whatever. Why would someone have network file system files or "files that appear in the root of a volume" inside their Flutter project? If they have, well it's there own fault...

On the other side, with this `.gitignore` we miss actual Flutter-related things that DO matter, like for example https://github.com/flutter/flutter/blob/master/.gitignore#L73-L101

Reply:
The main package doesn’t have `ios/` or `android/` folder. If you need it in the `example/`, you can add another `.gitignore` inside it for the additional rules needed for the `example/`. It can inherit from the main `.gitignore`.